src/@ionic-native/plugins/tealium/index.ts

```diff
@@ -1,217 +1,367 @@
-import { Cordova, IonicNativePlugin, Plugin } from '@ionic-native/core';
 import { Injectable } from '@angular/core';
+import { Plugin, Cordova, CordovaProperty, CordovaInstance, InstanceProperty, IonicNativePlugin } from '@ionic-native/core';
+import { Observable } from 'rxjs';
 
-export interface TealConfig {
-  /**
-   * Your Tealium account name
-   */
-  account: string;
-  /**
-   * Your Tealium profile name
-   */
-  profile: string;
-  /**
-   * Your Tealium environment name (dev, qa, prod)
-   */
-  environment: string;
-  /**
-   * Arbitrary instance name string. Must be consistent for all calls to Tealium API.
-   */
-  instance: string;
-  /**
-   * Enables ("true") or disables ("false") lifecycle reporting. Default true if omitted.
-   */
-  isLifecycleEnabled?: string;
-  /**
-   * Not usually required. Sets a custom URL override for dispatches to UDH.
-   */
-  collectDispatchURL?: string;
-  /**
-   * Your Tealium UDH profile. Only active if you do not have collectDispatchURL set.
-   */
-  collectDispatchProfile?: string;
-  /**
-   * Enables ("true") or disables ("false") Crash Reporter module for Android. Default false if omitted.
-   */
-  isCrashReporterEnabled?: string;
+export enum Collectors {
+	AppData = 'AppData',
+	Connectivity = 'Connectivity',
+	DeviceData = 'DeviceData',
+	Lifecycle = 'Lifecycle',
+}
+
+export enum Dispatchers {
+	Collect = 'Collect',
+	TagManagement = 'TagManagement',
+	RemoteCommands = 'RemoteCommands',
 }
 
+export enum Expiry {
+	forever = 'forever',
+	untilRestart = 'untilRestart',
+	session = 'session',
+}
+
+export enum ConsentPolicy {
+	ccpa = 'ccpa',
+	gdpr = 'gdpr',
+}
+
+export interface TealiumDispatch {
+	dataLayer: Map<string, any>;
+	type: string;
+	toJson(): string;
+}
+
+export class TealiumView implements TealiumDispatch {
+	public type: string = 'view';
+	constructor(public viewName: string, public dataLayer: Map<string, any>) {}
+	toJson() {
+		let dictionary: any = {};
+		dictionary['type'] = this.type;
+		dictionary['dataLayer'] = {};
+    this.dataLayer.forEach((k, v) => {
+      dictionary['dataLayer'][k] = v;
+    });
+		dictionary['type'] = this.type;
+		dictionary['viewName'] = this.viewName;
+		return JSON.stringify(dictionary);
+	}
+}
+
+export class TealiumEvent implements TealiumDispatch {
+	public type: string = 'event';
+	constructor(public eventName: string, public dataLayer: Map<string, any>) {}
+	toJson() {
+		let dictionary: any = {};
+		dictionary['type'] = this.type;
+		dictionary['dataLayer'] = {};
+    this.dataLayer.forEach((k, v) => {
+      dictionary['dataLayer'][k] = v;
```

Comment:
Right now with the current code, it is making the key as value and vice versa. 
It should be like this for both Event and View. 
```
this.dataLayer.forEach((value, key) => {
      dictionary['dataLayer'][key] = value;
});
```